server/src/main/java/org/elasticsearch/action/support/CancellableFanOut.java

```diff
@@ -122,7 +136,7 @@ public String toString() {
                 }
 
                 // Process the item, capturing a ref to make sure the outer listener is completed after this item is processed.
-                sendItemRequest(item, ActionListener.releaseAfter(itemResponseListener, refs.acquire()));
+                ActionListener.run(ActionListener.releaseAfter(itemResponseListener, refs.acquire()), l -> sendItemRequest(item, l));
```

Comment:
This I understand more as it's useful to capture any exceptions of the sendItemRequest that were not previously captured, and you also add a test for it.